### PR TITLE
[CodeMirror] UI-Refresh, unit tests fix, and  3.02 update

### PR DESCRIPTION
**Pull request by douglasduteil**
Hi @ProLoser 

Inspired by your last commits #316 I fixed my commented unit tests, added your ui-refresh and updated the CodeMirror version to 3.02.

### CODE REVIEW

**Comment by ProLoser** [comment]
Very cool. So this is an alternative to #316?

File: modules/directives/codemirror/test/codemirrorSpec.js [hunk 1]
```diff
@@ -2,135 +2,157 @@
...
+		});
+	});
+
+	describe('when the model changes', function () {
+		it('should update the IDE', function () {
+			var element = $compile('<textarea ui-codemirror ng-model="foo"></textarea>')(scope);
+			scope.foo = 'bar';
+			scope.$apply();
+			$timeout.flush();
+			expect($.trim(element.siblings().text())).toBe(scope.foo);
```
Comment (ProLoser) [inline modules/directives/codemirror/test/codemirrorSpec.js:120]: When I test locally this seems to be failing :(

**Comment by ProLoser** [comment]
Looks like travis only ran 1 test (need to open a bug on testacular about this).

Go to [travis](https://travis-ci.org/douglasduteil/angular-ui/builds/4686364), login with your github account, and in the dropdown menu of the build choose 'restart build'. Then scroll to the bottom of the build when it's done and confirm all tests ran. I don't get why when testacular does this and it drives me fucking nuts.

**Comment by douglasduteil** [comment]
K
I fixed some minor bug and now [Travis do all the tests](https://travis-ci.org/douglasduteil/angular-ui/builds/4696313) :)

It was there since the [90th travis build](https://travis-ci.org/angular-ui/angular-ui/builds/2993983) (the one that [changes the Testacular version](https://github.com/angular-ui/angular-ui/commit/0eac4de11d8cd5d9777df43adaa74c1f3c91ec1c) from "~0.2.0" to "~0.4.0").

**Comment by ProLoser** [comment]
There are merge conflicts with master. Additionally, I am getting lots of wonderful problems with my own tests so I may have to let someone else handle this.

Can you pull the latest ui/master version and push the conflict resolution for me? Perhaps I'm resolving things incorrectly.

Also, I know there was an (unrelated) discussion about using elm.type vs elm.tagName.toLowerCase() but since type is always available for textarea and since this directive will never be used with any other DOM element, can you revert it back to using 'type'?

Sorry for all the hassle lol. We are ALMOST THERE!

**Comment by douglasduteil** [comment]
I tried to rebase my repo on  **angular-ui/angular-ui** remote master branch . Is it better ?

**Pull request merged**